src/inmanta/tracing.py

```diff
@@ -19,21 +19,79 @@
 import contextlib
 import logging
 import os
-from typing import Any, Callable, ContextManager, LiteralString, Mapping, ParamSpec, Sequence, TypeVar
-
-import logfire
-import logfire._internal.config
-import logfire.integrations
-import logfire.integrations.pydantic
-from logfire import LevelName, LogfireSpan
-from logfire._internal.main import NoopSpan
-from logfire.propagate import ContextCarrier
+from typing import Any, Callable, ContextManager, Literal, LiteralString, Mapping, ParamSpec, Sequence, TypeVar
+
 from opentelemetry.instrumentation.asyncpg import AsyncPGInstrumentor
 
 LOGGER = logging.getLogger("inmanta")
 
-# Make sure we don't get warnings when it is off
-logfire._internal.config.GLOBAL_CONFIG.ignore_no_config = True
+# We need this early to make @instrument work
+enabled = os.getenv("LOGFIRE_TOKEN", None) is not None
+if enabled:
+    import logfire
+    import logfire._internal.config
+    import logfire.integrations
+    import logfire.integrations.pydantic
+    from logfire import LevelName, LogfireSpan
+    from logfire._internal.main import NoopSpan
+    from logfire.propagate import ContextCarrier
+
+    # Make sure we don't get warnings when it is off
+    logfire._internal.config.GLOBAL_CONFIG.ignore_no_config = True
+else:
```

Comment:
I'm a bit on the fence if this or the try except should be better. Any Opinion?

I'm a bit worried that with a `try except` approach, we try to enable logging even though the dependencies are not installed. We could still assert that the env var is not set if the library is not present

Reply:
Change to try and except, we should not crash if this token was used for something else